javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistTypeDeclarationAdapter.java

```diff
@@ -70,7 +70,7 @@ private void collectDeclaredFields(CtClass ctClass, List<ResolvedFieldDeclaratio
 
   public List<ResolvedTypeParameterDeclaration> getTypeParameters() {
     if (null == ctClass.getGenericSignature()) {
-      return Collections.emptyList();
+      return new ArrayList<>();
```

Comment:
Again, why? We are not trying to modify this list, right?

Reply:
ok, this is probably not needed here, but only in JavassistMethodDeclarationAdapter. I'll check if junit tests are still working when I revert this change.

Reply:
I've reverted the new ArrayList creation here in this class, it was not necessary.